tgui/packages/tgui-bench/index.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file
+ * @copyright 2021 Aleksej Komarov
+ * @license MIT
+ */
+
+const fs = require('fs');
+const path = require('path');
+const { exec } = require('child_process');
+const { fastify } = require('fastify');
+
+process.chdir(__dirname);
+
+const sleep = (time) => new Promise((resolve) => setTimeout(resolve, time));
+
+const setup = async () => {
+  const server = fastify();
+
+  let hasResponded = false;
+
+  let assets = '';
+  assets += `<script>\n`;
+  assets += `Byond.loadJs('tgui-bench.bundle.js');\n`;
+  assets += `Byond.loadCss('tgui-bench.bundle.css');\n`;
+  assets += `</script>\n`;
+
+  const publicDir = path.resolve(__dirname, '../../public');
+  const page = fs.readFileSync(path.join(publicDir, 'tgui.html'), 'utf-8')
+    .replace('<!-- tgui:assets -->\n', assets);
+
+  server.register(require('fastify-static'), {
+    root: publicDir,
+  });
+
+  server.get('/', async (req, res) => {
+    return res.type('text/html').send(page);
+  });
+
+  server.post('/message', async (req, res) => {
+    if (!hasResponded) {
+      process.stdout.write('\n');
+      hasResponded = true;
+    }
+    const { type, ...rest } = req.body;
+    if (type === 'suite-start') {
+      console.log(`=> Test '${rest.file}'`);
+      return res.send();
+    }
+    if (type === 'suite-cycle') {
+      console.log(rest.message);
+      return res.send();
+    }
+    if (type === 'suite-complete') {
+      console.log(rest.message);
+      return res.send();
+    }
+    if (type === 'finished') {
+      await res.send();
+      process.exit(0);
+    }
+    // Unhandled message
+    console.log(req.body);
+    return res.send();
+  });
+
+  try {
+    await server.listen(3002, '0.0.0.0');
+  }
+  catch (err) {
+    console.error(err);
+    process.exit(1);
+  }
+
+  if (process.platform === 'win32') {
+    exec(`start "" "iexplore" "http://127.0.0.1:3002"`);
+  }
+
+  console.log('Waiting for Internet Explorer to respond.');
+  for (let i = 0; i < 60; i++) {
+    await sleep(1000);
```

Comment:
wouldn't these being some sort of constants be ideal? it'd make it easier to configure as you wouldn't have to go to this specific point in code if you want to change how often it checks for internet explorer's response or how often it does it before timing out.

Reply:
Can you think of a case where you might want to configure this? 60 seconds is plenty and barely reaches that, usually starts in split second.

Reply:
well, it's more to just have the option of being able to modify it instead of having to hardcode it. I can't think of any scenarios right now where you'd want to actually change this. The benefits is that it lets anyone see when it times out more easily as it'd be in some sort of constants/config file.

I don't think it's necessary, but it just generally makes your code more configurable if people can modify values like these somewhere, even if there might not be any reason to right now.

Reply:
hell, you could even place it at the top of the file, it's easier than having to go into the code though